kcidb/monitor/output.py

```diff
@@ -184,9 +185,11 @@ def render(self):
             subject = subject[:NOTIFICATION_MESSAGE_SUBJECT_MAX_LEN - 1] + "✂️"
             LOGGER.warning("Subject is %s characters too long, truncated",
                            subject_extra_characters)
-        assert NOTIFICATION_MESSAGE_SUBJECT_RE.fullmatch(subject), \
-            f"Subject is invalid, must match " \
-            f"{NOTIFICATION_MESSAGE_SUBJECT_RE.pattern} regex"
+        if not NOTIFICATION_MESSAGE_SUBJECT_RE.fullmatch(subject):
+            subject = re.sub('[^A-Za-z0-9]+', ' ', subject).strip()
```

Comment:
This would replace any number of consecutive non-Latin and non-digit characters with a single space. This means that text in any language not using Latin characters will be replaced with a single space, which we don't want to do. This would also make it difficult to determine how many characters were there, and make debugging harder.

Instead, we should replace exactly the inverse of permitted characters, and replace them one-to-one.

Also, it's better to use a character which would clearly signify that something was replaced. There's the old "[substitute character](https://en.wikipedia.org/wiki/Substitute_character)" in ASCII and Unicode, but it's also a control character and could be interpreted differently. There's the Unicode "[replacement character](https://en.wikipedia.org/wiki/Specials_(Unicode_block)#Replacement_character)", but it normally signifies that there was something wrong in the Unicode encoding. I couldn't find anything better than the "[uncertainty sign](https://codepoints.net/U+2BD1?lang=en)" with an uncertain (ha-ha) purpose, which looks like the "replacement character", but is white rather than black. It will have to do. Could you use that instead as the replacement?

To make sure our "correct" and "incorrect" character patterns are always exactly the inversion of each other, we can generate one from another.

We can e.g. define a new variable called `NOTIFICATION_MESSAGE_SUBJECT_INVALID_CHARS` specifying invalid characters, and set it to `\x00-\x1f\x7f`. Then we can generate both regular expressions from that, to make sure they are always the opposites of each other. Can you do that?